datanode/src/main/java/com/odysseusinc/arachne/datanode/dto/datasource/validation/context/CredentialsValidationContextBuilder.java

```diff
@@ -42,13 +42,20 @@ public CredentialsValidationContextBuilder withUsername(String fieldName, String
         return this;
     }
 
+
     public CredentialsValidationContextBuilder withKeyfile(String fieldName, Object value) {
 
         contextInstance.setKeyfileField(fieldName);
         contextInstance.setKeyfile(value);
         return this;
     }
 
+    public CredentialsValidationContextBuilder usingKeyFile(boolean useKeyFile) {
```

Comment:
Why need to add this? The boolean flag can be just set from the original setter.

Reply:
Yeah, but that is a clumsy builder. It should be completely revised. At the moment, I’m trying to make it consistent.